### ISSUE TITLE
Request for clarification/documentation

### ISSUE THREAD

**Issue by DavidJFelix** [tj/luna#87]
Does Luna have null?
Does Luna have the concept of tagged unions/ variant types?
Can you pass a function as a parameter to another function, if so, what is the type declaration of the function?

Thanks.

**Comment by buddywithgol** [comment]
I agree with you that there need to be documentation and language specification.

**Comment by mistryalok** [comment]
Agree, we need some kind of documentation, Readme is not enough for that.

**Comment by aisk** [comment]
I think luna is under very alpha development stage, everything may changed quickly, so the document is too early to start. The first goal is to make luna can run real code and we can attempt to write it and found what is missing / bad and improve it.

And I think we can discuss the problems in the issues if anyone thought of it.

And about your question, in my opinion (I'm not the luna developer 😁), I don't like null, and the Option<T> type looks cool. And function as first citizen is cool I and I think we must support it.
